### PR TITLE
Add typehints and fstrings to decode_cmd.py

### PR DESCRIPTION
**Pull request by dscott90**

### CODE REVIEW

File: chipsec/utilcmd/decode_cmd.py [hunk 1]
```diff
@@ -54,32 +54,33 @@
...
         return False
 
-    def decode_types(self):
-        self.logger.log("\n<fw_type> should be in [ {} ]\n".format(" | ".join(["{}".format(t) for t in uefi_platform.fw_types])))
+    def decode_types(self) -> None:
+        self.logger.log(f'\n<fw_type> should be in [ {" | ".join([f"{t}" for t in uefi_platform.fw_types])} ]\n')
 
-    def decode_rom(self):
-        self.logger.log("[CHIPSEC] Decoding SPI ROM image from a file '{}'".format(self._rom))
+    def decode_rom(self) -> bool:
```
[the remark this answers is not in the document]
Comment (fishilico) [reply chipsec/utilcmd/decode_cmd.py:72]: This CodeQL notice is caused by the fact that a `return True` seems to be missing from the end of this function.

File: chipsec/utilcmd/decode_cmd.py [hunk 2]
```diff
@@ -54,32 +54,33 @@
 from chipsec.hal.spi_descriptor import get_spi_flash_descriptor, get_spi_regions, parse_spi_flash_descriptor
 from chipsec.hal.spi_uefi import decode_uefi_region
 from chipsec.hal.uefi import uefi_platform
+from typing import Optional
```
[the remark this answers is not in the document]
Comment (npmitche) [reply chipsec/utilcmd/decode_cmd.py:57]: Should clean this up.

**Review by npmitche — CHANGES_REQUESTED** [review]
Clean up codeQL alerts.

File: chipsec/utilcmd/decode_cmd.py [hunk 3]
```diff
@@ -58,28 +58,29 @@
...
         f = read_file(self._rom)
         if not f:
             return False
         (fd_off, fd) = get_spi_flash_descriptor(f)
         if (-1 == fd_off) or (fd is None):
-            self.logger.log_error("Could not find SPI Flash descriptor in the binary '{}'".format(self._rom))
+            self.logger.log_error(f'Could not find SPI Flash descriptor in the binary \'{self._rom}\'')
             self.logger.log_information("To decode an image without a flash decriptor try chipsec_util uefi decode")
             return False
+        return True
```
Comment (fishilico) [inline chipsec/utilcmd/decode_cmd.py:81]: This `return True` needs to be moved a few lines down, where the function ends (line 118)

**Review by fishilico — CHANGES_REQUESTED** [review]

**Review by fishilico — APPROVED** [review]

**Review by npmitche — APPROVED** [review]

**Pull request merged**